src/v0/destinations/twitter_ads/transform.js

```diff
@@ -0,0 +1,227 @@
+  let authHeader = getAuthHeaderForRequest(request, oAuthObject).Authorization;
+  if(message.properties.testModeEnable === true) {
+    authHeader = authHeaderConstant;
+  }
+  response.headers = {
+    Authorization: authHeaderConstant,
+    'Content-Type': JSON_MIME_TYPE,
+  };
+  return response;
+}
+
+function prepareUrl(message, destination) {
+  const pixelId = message.properties.pixelId || destination.Config.pixelId;
+  return `${BASE_URL}/${pixelId}`;
+}
+
+function populateEventId(event, requestJson, destination) {
+
+  const eventNameToIdMappings = destination.Config.twitterAdsEventNames;
+  let eventId = "";
+
+  if (eventNameToIdMappings) {
+    eventNameToIdMappings.forEach(obj => {
+      if (obj.rudderEventName && obj.rudderEventName.trim() && obj.rudderEventName.trim().toLowerCase() === event.toString().toLowerCase()) {
+        eventId = obj.twitterEventId;
+      }
+    });
+  }
+
+  if(!eventId) {
+    throw new ConfigurationError(`[TWITTER ADS]: Event - '${event}' do not have a corresponding eventId in configuration. Aborting`);
+  }
+
+  return eventId;
+}
+
+// process track call
+function processTrack(message, metadata, destination) {
+
+  const requestJson = constructPayload(message, mappingConfig[ConfigCategories.TRACK.name]);
+
+  requestJson.event_id = requestJson.event_id || populateEventId(message.event, requestJson, destination);
+
+  requestJson.conversion_time = isDefinedAndNotNull(requestJson.conversion_time)
+    ? requestJson.conversion_time : message.timestamp;
+
+  const identifiers = [];
+
+  if (message.properties.email) {
+    let email = message.properties.email.trim();
+    if (email) {
+      email = email.toLowerCase();
+      identifiers.push({hashed_email: sha256(email)})
+    }
+  }
+
+  if (message.properties.phone) {
+    let phone = message.properties.phone.trim();
+    if (phone) {
+      identifiers.push({hashed_phone_number: sha256(phone)})
+    }
+  }
+
+  if (message.properties.twclid) {
+    identifiers.push({twclid: sha256(message.properties.twclid)});
+  }
+
+  if (requestJson.contents) {
+    const transformedContents = [];
+    requestJson.contents.forEach(obj => {
+      const transformedObj = {};
+      if (obj.id) {
+        transformedObj.content_id = obj.id;
+      }
+
+      if (obj.groupId) {
+        transformedObj.content_group_id = obj.groupId;
+      }
+
+      if (obj.name) {
+        transformedObj.content_name = obj.name;
+      }
+
+      if (obj.price) {
+        transformedObj.content_price = parseFloat(obj.price);
+      }
+
+      if (obj.type) {
+        transformedObj.content_type = obj.type;
+      }
+
+      if (obj.quantity) {
+        transformedObj.num_items = parseInt(obj.quantity, 10);
+      }
+
+      if (Object.keys(transformedObj).length > 0) {
+        transformedContents.push(transformedObj);
+      }
+    });
+    requestJson.contents = transformedContents;
+
+    if(transformedContents.length === 0) {
+      delete requestJson.contents;
+    }
+  }
```

Comment:
```suggestion
  if (requestJson.contents) {
    const transformedContents = requestJson.contents.map(obj => ({
        ...(obj.id && { content_id: obj.id }),
        ...(obj.groupId && { content_group_id: obj.groupId }),
        ...(obj.name && { content_name: obj.name }),
        ...(obj.price && { content_price: parseFloat(obj.price) }),
        ...(obj.type && { content_type: obj.type }),
        ...(obj.quantity && { num_items: parseInt(obj.quantity, 10) })
      })).filter(tfObj => Object.keys(tfObj).length > 0);
    if (transformedContents.length > 0) {
      requestJson.contents = transformedContents;
    }
  }
```
This feels like is more readable & clean